lib/ruby_indexer/lib/ruby_indexer/declaration_listener.rb

```diff
@@ -751,6 +753,41 @@ def handle_module_operation(node, operation)
       end
     end
 
+    sig { params(node: Prism::CallNode).void }
+    def handle_module_function(node)
+      arguments_node = node.arguments
+      return unless arguments_node
+
+      name_argument = arguments_node.arguments.first
+      return unless name_argument.is_a?(Prism::SymbolNode)
+
+      method_name = name_argument.value.to_s
+      owner_name = @owner_stack.last&.name
+      return unless owner_name
+
+      entries = @index.resolve_method(method_name, owner_name)
+      return unless entries
+
+      entries.each do |entry|
+        entry_owner_name = entry.owner&.name
+        next unless entry_owner_name
+
+        entry.visibility = Entry::Visibility::PRIVATE
+
+        singleton = @index.existing_or_new_singleton_class(entry_owner_name)
+        @index.add(Entry::Method.new(
+          method_name,
+          @file_path,
+          Location.from_prism_location(node.location, @code_units_cache),
+          Location.from_prism_location(T.must(node.message_loc), @code_units_cache),
+          collect_comments(node),
+          [],
```

Comment:
Here we need to pass the same signatures as the original method, otherwise features like signature help and hover will not show the correct signature.

```suggestion
          entry.signatures,
```

Reply:
I am confused, as I moved the `@index.add` out of the loop. Where can I get the correct `signatures` and pass it to the `@index.add`?

Reply:
That's a great question. My original comment was not correct.

Imagine we have this scenario where a method is overridden with an incompatible signature. We can't know which override wins, because there's no way to confidently determine the load order of the files (due to autoload and dynamic requires).

```ruby
# file1.rb
# If this file is loaded last, then bar accepts one parameter named `a`
class Foo
  def bar(a)
  end
end

# file2.rb
# If this file is loaded last, then bar accepts two parameters named `b` and `c`
class Foo
  def bar(b, c)
  end
end

# In this scenario, invoking `@index.resolve_method("bar", "Foo")` will return two
# entries. One for each `bar` declaration
```

Since we cannot know which override wins, I think the best decision is to add singleton entries for each override (i.e.: move the `index.add` inside of the loop and use the same signature from the original instance method).

Reply:
Hey, Thank you for bringing this up. I've never considered this case, but I'm glad we know it could happen.